tldw_Server_API/app/core/TTS/tts_validation.py

```
@@ -654,6 +664,31 @@ def validate_request(self, request: TTSRequest, provider: Optional[str] = None)
+                if isinstance(extras, dict):
+                    ref_text = (
+                        extras.get("reference_text")
+                        or extras.get("ref_text")
+                        or extras.get("voice_reference_text")
+                    )
+                clone_requested = bool(request.voice_reference) or voice.lower() == "clone" or is_custom_voice
+                if is_custom_voice and not request.voice_reference:
+                    raise TTSInvalidVoiceReferenceError(
+                        "OmniVoice custom: voices require a resolved voice_reference before provider validation",
+                        provider=provider,
+                    )
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)